src/ZuluSCSI_cdrom.cpp

```diff
@@ -492,23 +514,20 @@ static void doReadTOC(bool MSF, uint8_t track, uint16_t allocationLength)
         if (firsttrack < 0) firsttrack = trackinfo->track_number;
         lasttrack = trackinfo->track_number;
 
-        if (track == 0 || track == trackinfo->track_number)
+        if (track <= trackinfo->track_number)
```

Comment:
Ah, indeed a bug. I missed that track specifies starting track for reporting and not the only track for reporting.